tools/rccl_replayer/rcclReplayer.cpp

```diff
@@ -0,0 +1,379 @@
+
+        gc.isValid = true;
+
+        if (mpiRank == 0) {
+            printf("GroupCall %d\n", i);
+            printf(" - OpCount: %d\n", gc.opCount);
+        }
+
+        for (auto rd : gc.rankData) {
+            if (mpiRank == 0) {
+                printf("  - Rank %02d: comm %s\n", rd.first, rd.second.comm.c_str());
+            }
+
+            for (int task = 0; task < rd.second.tasks.size(); task++) {
+                TaskInfo ti = rd.second.tasks[task];
+                const char* funcName;
+
+                if (ti.funcType == ncclCollSend || ti.funcType == ncclCollRecv)
+                    funcName = "Send/Recv";
+                else 
+                    funcName = ncclFuncNames[ti.funcType];
+
+                std::tuple<const char*, size_t, int, int> key(funcName, ti.count, ti.datatype, ti.op);
+
+                if (mpiRank == 0) {
+                    printf("  - Task %02d: %32s inPlace=%d count=%lu datatype=%d op=%d root=%d\n",
+                        task, funcName, ti.inPlace, ti.count, ti.datatype, ti.op, ti.root);
+                }
+
+                auto& rankVector = arrivalCounter[key];
+                if (rankVector.size() < numGlobalRanks) {
+                    rankVector.resize(numGlobalRanks);
+                }
+
+                // rankVector<int> in arrivalCount represents the rank information
+                // Count the number of tasks that are going to be executed by each rank. This is to validate the group call later on.
+                // Nom-Send/Recv rank counts (rankVector<int> elements) should be equal at the end, and for Send/Recv, all the elements of rankVector<int> should be equal to 0
+                if (strcmp(ncclFuncNames[ti.funcType], "Recv") == 0) {
+                    rankVector[ti.root]--;
+                } else {
+                    rankVector[rd.first]++;
+                }
+            }
+        }
+
+        // Iterate through the map variable and report/validate the results
+        for (const auto& e : arrivalCounter) {
+            int maxVal;
+            bool validateRanks = true;
+            
+            bool isp2p = (strcmp(std::get<0>(e.first), "Send/Recv") == 0);
+            if (!isp2p) maxVal = *std::max_element(e.second.begin(), e.second.end());
+            
+            // Validate all the ranks have required amount of collective call (task)
+            for (int i = 0; i < e.second.size(); i++) {
+                if (e.second[i] != (isp2p ? 0 : maxVal)) {
+                    std::string warning = (isp2p ? (e.second[i] > 0 ? "[WARN] Missing Recv" : "[WARN] Missing Send") : "[WARN] Missing " + std::string(std::get<0>(e.first)) ) 
+                            + " count=" + std::to_string(std::get<1>(e.first)) + " datatype=" + std::to_string(std::get<2>(e.first)) + " op=" + std::to_string(std::get<3>(e.first)) + " at rank [" + std::to_string(i) + "]";
+                    if(mpiRank == 0) printf("%s\n", warning.c_str());
```

Comment:
This is hard to follow with all the get<>.
Given e, just create named references to the various tuple fields, and use that in your code to improve readability
e.g.:
char* funcName = std::get<0>(e.first);
size_t count          = std::get<1>(e.first);
...